### ISSUE TITLE
Syntax highlighting for C++ breaks after a specific printf statement

### ISSUE THREAD

**Issue by cbuchner1** [Microsoft/vscode#4644]
- VSCode Version:
0.10.11
- OS Version:
Kubuntu 16.04 LTS Developer branch

Steps to Reproduce:

1. I have a .cc file with the following contents.

   #if B4G_DEBUG_CHECK
    fprintf(stderr,"num_candidate_ret=%d:", num_candidate);
    for(int i=0;i<num_candidate;++i)
        fprintf(stderr,"%d,",user_candidate[i]);
    fprintf(stderr,";");
    #endif

2. Observe that syntax highlighting for any statements following the  fprintf(stderr,";"); assumes that the following lines are contained within the double quote and they show up in red. Apparently the ; inside the double quote is misinterpreted as the end of a statement. The "); on the same line is already shown in all red color, and all following lines are too - unless they contain another double quote. In this case the string inside the double quote is shown as if it was a code statement.
![snapshot1](https://cloud.githubusercontent.com/assets/416723/14019003/e6c4c13e-f1d0-11e5-9160-c6527eaec4b6.png)

Partial screenshot attached.

**Comment by cbuchner1** [comment]
Updating this ticket with the additional information that any semicolon within a string will do that.
I found the issue also to be triggered by statements such as 

sprintf(options, "STYLE=Keramik;TITLE=%s;THEME=%s", ...);

**Comment by CBielstein** [comment]
I'm having a similar problem, but it's with brackets inside of quotation marks. In my file, I have fprintf(fp, "some stuff"); It doesn't highlight the string at all. But if I put fprintf(fp, "{some stuff}"); then it highlights starting at the closing quotation mark down through the rest of the file. Shown in the attached screenshot. It happens with any of an opening, closing, or matching brackets.
![vscodebug](https://cloud.githubusercontent.com/assets/3268813/14035605/8b22e460-f1ed-11e5-866e-50c58cf8d959.PNG)

**Comment by Eyas** [comment]
Another one that seems related (strings in function calls):
```c++
void main(O obj) {
    LOG_INFO("not hilighted as string");
    LOG_INFO(obj << ", even worse; " << obj.x << " check this out.");
    // everything from this point on is interpeted as a string literal...
    O x;
    std::unique_ptr<O> o(new O);
    // sadness.
}
```

Shown as:

![image](https://cloud.githubusercontent.com/assets/554424/14084492/f6a8db4c-f4e8-11e5-837b-b97dbf939e7c.png)

**Comment by aeschli** [comment]
these issues got fixed by  2d56597

**Comment by aeschli** [comment]
duplicate of #4769
